src/components/fabSpeedDial/fabSpeedDial.js

```diff
@@ -0,0 +1,138 @@
+(function() {
+  'use strict';
+
+  angular
+    .module('material.components.fabSpeedDial', [])
+    .directive('mdFabSpeedDial', MdFabSpeedDialDirective);
+
+  /**
+   * @ngdoc directive
+   * @name mdFabSpeedDial
+   * @module material.components.fabSpeedDial
+   *
+   * @restrict E
+   *
+   * @description
+   * The `<md-fab-speed-dial>` directive is used present a series of popup elements (usually `<md-button>`s) for
+   * quick access to common actions.
+   *
+   * @usage
+   * <hljs lang="html">
+   * <md-fab-speed-dial direction="up">
+   *   <md-fab-trigger>
+   *     <md-button><md-icon icon="/img/icons/plus.svg"></md-icon></md-button>
+   *   </md-fab-trigger>
+   *
+   *   <md-fab-actions>
+   *     <md-button><md-icon icon="/img/icons/user.svg"></md-icon></md-button>
+   *     <md-button><md-icon icon="/img/icons/group.svg"></md-icon></md-button>
+   *   </md-fab-actions>
+   * </md-fab-speed-dial>
+   * </hljs>
+   *
+   * @param {string=} md-direction From which direction you would like the speed dial to appear relative to the
+   * trigger element.
+   * @param {expression=} md-open Programmatically control whether or not the speed-dial is visible.
+   */
+  function MdFabSpeedDialDirective($timeout) {
+
+    return {
+      restrict: 'E',
+
+      scope: {
+        direction: '@?mdDirection',
+        isOpen: '=?mdOpen'
+      },
+
+      controller: controller,
+
+      compile: function($element) {
+        // Wrap every child in a new div and add a class that we can scale independently
+        var children = $element.find('md-fab-actions').children(),
+            child, i;
+
+        for (i = 0; i < children.length; i++) {
+          child = children.eq(i);
+
+          child.wrap('<div class="md-fab-action-item">');
+        }
+
+        // Return our link function
+        return {
+          post: link
+        }
+      }
+    };
+
+    function controller($scope, $element) {
+      var vm = this;
+
+      vm.scope = $scope;
+      vm.element = $element;
+
+      // Set the default direction to 'down' if none is specified
+      if (typeof $scope.direction === 'undefined') {
+        $timeout(function() {
+          $scope.direction = 'down';
+        });
+      }
+
+      // Set the default to be closed
+      if (typeof $scope.isOpen === 'undefined') {
+        $timeout(function() {
+          $scope.isOpen = false;
+        });
+      }
+
+      vm.focus = function() {
+        $timeout(function() {
+          vm.scope.isOpen = true;
+        });
+      };
+
+      vm.blur = function() {
+        $timeout(function() {
+          vm.scope.isOpen = false;
+        });
+      };
+
+    }
+
+    function link($scope, $element, $attrs, ctrl) {
+      var i, trigger, child, children, focusable;
```

Comment:
@jelbourn I would disagree with this - it might be personal preference, but I much prefer having a set list up top so that I know the variables being used.